components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/authz/AuthorizationHandlerManager.java

```diff
@@ -208,6 +218,199 @@ private OAuth2AuthorizeRespDTO validateAuthzRequest(OAuth2AuthorizeReqDTO authzR
+            // OIDC scopes are not validated in the scope validation process. Hence, Removing OIDC scopes from the
+            // requested scopes by the app.
+            String[] oidcRemovedScopes = OAuth2Util.removeOIDCScopesFromRequestedScopes(authzReqMsgCtx
+                    .getAuthorizationReqDTO().getScopes());
+            authzReqMsgCtx.getAuthorizationReqDTO().setScopes(oidcRemovedScopes);
+        } catch (IdentityOAuthAdminException e) {
+            throw new IdentityOAuth2Exception("Error occurred while validating requested scopes.", e);
+        }
+        // Get scopes that specified in the allowed scopes list.
+        List<String> requestedAllowedScopes = getAllowedScopesFromRequestedScopes(authzReqMsgCtx);
+        // If it is management app, we validate internal scopes in the requested scopes.
+        String[] authorizedInternalScopes = new String[0];
+        boolean isManagementApp = isManagementApp(authzReqMsgCtx.getAuthorizationReqDTO());
+        if (isManagementApp) {
+            if (log.isDebugEnabled()) {
+                log.debug("Handling the internal scope validation.");
+            }
+            authorizedInternalScopes = getAuthorizedInternalScopes(authzReqMsgCtx);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("Skipping the internal scope validation as the application is not" +
+                        " configured as Management App");
+            }
+        }
+        // Remove the internal scopes from requested scopes for further validation.
+        removeInternalScopesFromRequestedScopes(authzReqMsgCtx);
+        // Adding the authorized internal scopes to tokReqMsgCtx for any special validators to use.
+        authzReqMsgCtx.setAuthorizedInternalScopes(authorizedInternalScopes);
+        boolean isDropUnregisteredScopes = OAuthServerConfiguration.getInstance().isDropUnregisteredScopes();
+        if (isDropUnregisteredScopes) {
+            if (log.isDebugEnabled()) {
+                log.debug("DropUnregisteredScopes config is enabled. Attempting to drop unregistered scopes.");
+            }
+            dropUnregisteredScopeFromRequestedScopes(authzReqMsgCtx);
+        }
+        //Validate scopes using global scope validators.
+        boolean isValid = validateScopes(authzReqMsgCtx, authzHandler);
+        if (isValid) {
+            // Add authorized internal scopes to the request for sending in the response.
+            addAuthorizedInternalScopes(authzReqMsgCtx, authzReqMsgCtx.getAuthorizedInternalScopes());
+            // Add OIDC scopes back to the request.
+            addRequestedOIDCScopes(authzReqMsgCtx, requestedOIDCScopes);
+            // Add scopes that filtered from the allowed scopes list.
+            addAllowedScopes(authzReqMsgCtx, requestedAllowedScopes.toArray(new String[0]));
+        } else {
+            throw new IdentityOAuth2UnauthorizedScopeException(INVALID_SCOPE, "Requested scopes can not be validated.");
```

Comment:
`Requested scopes can not be validated.`

Shouldn't the error indicate the scope validation failed. Current message gives the feeling the something went wrong when we tried to validate the scopes